File: 70-climbing-stairs/climbing-stairs.py
from math import comb

class Solution:
    def climbStairs(self, n: int) -> int:
        total_ways = 0
        # Loop through possible counts of 2-steps
        for two_steps in range(n // 2 + 1):
            one_steps = n - 2 * two_steps
            # Calculate permutations using combinations formula
            total_ways += comb(one_steps + two_steps, two_steps)
        return total_ways

File: 70-climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int count = 0;
        int first = 0;
        int second = 1;
        int steps = 0;
        while(count < n){
            steps = first + second;
            first = second;
            second = steps;
            count++;
        }
        return steps;
    }
}